### ISSUE TITLE
Safari Synth plays only after second button click

### ISSUE THREAD

**Issue by Ksinia** [paulrosen/abcjs#328]
Hello!

Both [examples/basic-synth.html](https://github.com/paulrosen/abcjs/blob/master/examples/basic-synth.html) and [examples/full-synth.html](https://github.com/paulrosen/abcjs/blob/master/examples/full-synth.html) don't work for me in the same manner in Chrome and Safari.

[Example page with basic-synth](https://tunearch.org/wiki/TTA:Synth). (A side note, would be nice to have examples HTML files deployed somewhere on [abcjs.net](https://abcjs.net/)).

In Chrome the behavior is as expected: single click on `Activate Audio Context And Play` button starts the play and I hear the music.

In Safari (both mobile and desktop) I need to click `Activate Audio Context And Play`, `Stop Audio` and `Activate Audio Context And Play` again in order to hear the music.

I've tried to work this around in my application in multiple ways but have not found a way around it so far. I found that I need to pre-load the notes mp3 files (like `C2.mp3`) with the first button click calling `midiBuffer.init()`, and only on the second button click call to `midiBuffer.start()` will work and produce the sound.

/cc @yvesgurcan: I believe you might be experiencing the same problem in web-midi-player.

**Comment by paulrosen** [comment]
I can't reproduce this on my computer because I only have Safari 11. That works fine. I'm guessing you have Safari 13? In BrowserStack I can't get any audio to play so I can't tell.

**Comment by paulrosen** [comment]
I looked at the network tab in Safari 13 in BrowserStack and I see the mp3 files loading when the page loads. Do you see that? That is, open the devtools/network tab and reload the page. I see all the individual note files loading immediately.

**Comment by Ksinia** [comment]
Safari 13.0.5 (15608.5.11), and mp3 files are loading only after the first click on `Activate Audio Context And Play` button.

**Comment by paulrosen** [comment]
Thanks. I'll delay loading the mp3 until the user clicks.

**Comment by Ksinia** [comment]
Duplicating from #325 as it's more relevant here: GitHub master version of abcjs at the time of the comment (`npm install --save paulrosen/abcjs#master` 082e79c) doesn't work with Safari 13 at all: the mp3 files are loaded but never plays.

[The code I use for play](https://github.com/Ksinia/melodic-dictation-client/blob/c02f8210e1bc9d881109abe5b88e6e29ed38a8a3/src/components/MelodyListContainer.js#L11-L27).

**Comment by paulrosen** [comment]
I wasn't able to reproduce this on a iPhone with Safari 13.

I did realize something - I have not yet updated the examples in master so you would be using the old code if you are just opening `basic-synth.html`. And the example you posted in the first comment is using version 5.12. 

The easiest way to try the latest code is to remove the `-min` from "../bin/abcjs_basic_5.12.0.js" and from the command line run: `npm run build:basic`.

I'll soon be releasing this code, but I'm hoping to have this issue resolved first.

**Comment by Ksinia** [comment]
Unfortunately, it stopped working in Safari at all for me.

```
    "abcjs": {
      "version": "github:paulrosen/abcjs#36859117c5883581d992b65f013c53f6522c7c56",
      "from": "github:paulrosen/abcjs#master",
      "requires": {
        "abcjs": "6.0.0-beta.2",
        "midi": "git+https://github.com/paulrosen/MIDI.js.git#abcjs"
      }
    },
```

**Comment by paulrosen** [comment]
I already stumbled on the Firefox problem. It should be fixed in version "6.0.0-beta.1". 

I still am not seeing the problem in Safari - I think that points to whether Safari trusts the web site and will allow the AudioContext. At least, these problems usually boil down to that. I'll create a tiny test program that gives more diagnostics.

Thanks for sticking with this.

**Comment by yvesgurcan** [comment]
Hey @paulrosen, if it can help, I just fixed an issue with `AudioContext` and Safari. From what I understand, the problem is about the time between the click and the creation of the `AudioContext`. If it's too long, Safari will somehow not allow playback. If it can help, here is the relevant snippet of code that made it all work for Web MIDI Player: https://github.com/yvesgurcan/web-midi-player/blob/master/src/MidiPlayer.js#L118-L124. @Ksinia was able to test it and confirm that it works with Safari. The example can be tested here: https://midi.yvesgurcan.com/. Hope it helps!

**Comment by paulrosen** [comment]
Thanks! I'll look into it soon. I know that happens with Safari and recording - if you ask for permission a minute in advance it makes the user confirm a second time.

**Comment by Ksinia** [comment]
@paulrosen I've created [a test page](https://melodic-dictation.netlify.com/abcjs_play_test) for you to see what I am observing. Synth play works with Firefox and Chrome but not in Safari 13, neither desktop nor mobile.

I saw it working on Desktop safari occasionally one in 20 times but can't reproduce it now.

**Comment by paulrosen** [comment]
Hi, sorry for the delay - Mardi Gras is a pretty busy time for me so I'm just getting back to this. I also created an example here: https://codepen.io/paul314259/pen/ExjWJJP and I observe the problem. I'll get that working and release a new beta when I do.

**Comment by paulrosen** [comment]
The AudioContext was being created in the "suspended" state. @yvesgurcan is on the mark. The odd thing is there isn't very much time between clicking and creation - it is close to the top of the hander. Also, I can just immediately resume it, so I'm not sure what suspending it protects anyone from. So @Ksinia , can you confirm that the codepen I listed above now works for you? I only have one iOS device to test on.

**Comment by Ksinia** [comment]
Hi @paulrosen,
It works for me both on Safari desktop and mobile. Thanks a lot!

**Comment by yvesgurcan** [comment]
Woohoo! Glad to hear you could fix it. Yeah, I think the biggest problem for me is that there is no indication on Safari's part that it stops the playback. Browsers like Chrome would output a message in the console for a similar case.

**Comment by paulrosen** [comment]
Fixed in the 6.0.0 beta

**Comment by josyjanga** [comment]
Hi Paul,
sounds like this issue got fixed, which is great! I think the following problem relates to this issue:
When I click the play button at his example https://configurator.abcjs.net/audio/ on an iPad or iPhone (I tested different mobile apple devices with different versions always with Browser Safari), the sound stutters, it doesn't play smoothly. I made a video so you can see and hear it: 
https://guglerat-my.sharepoint.com/:v:/g/personal/elisabeth_zeilinger_gugler_at/EY6YQiUHjUtHtM7sAouyuwoB8HaCSpj45yq0b22pCimpXw?e=dWGnns

Do you know this problem and how to fix this?
Which version is the configurator using?

Thanks and keep up the great work!

**Comment by paulrosen** [comment]
Hi, the configurator is way behind in versions - it is definitely still useful but I'm trying to roll its functionality in the regular documentation.

I hope the newer version works better than that - I think it does. If you play it a second time does it act better?

Try the same with https://editor.drawthedots.com and see how that does. That's not the very latest but it is a lot newer. I'll be updating that again pretty soon.

**Comment by josyjanga** [comment]
Thanks Paul! With the newer version here https://editor.drawthedots.com it works fine on apple devices.
